frontend/src/components/questions/QuestionDeleteIconButton.tsx

```diff
@@ -0,0 +1,41 @@
+import { DeleteIcon } from '@chakra-ui/icons';
+import { Tooltip, IconButton } from '@chakra-ui/react';
+import React from 'react';
+import QuestionsAPI from '../../api/questions/questions';
+
+interface QuestionDeleteIconButtonProps {
+  questionId: number;
+  onDelete: (questionId: number) => void; // Callback function for handling deletion
+}
+
+const QuestionDeleteIconButton: React.FC<QuestionDeleteIconButtonProps> = ({
+  questionId,
+  onDelete,
+}: QuestionDeleteIconButtonProps) => {
+  const handleDelete = (): void => {
+    new QuestionsAPI()
+      .deleteQuestion(questionId)
+      .then(() => {
+        // Call the onDelete callback when the delete is successful
+        onDelete(questionId);
+      })
+      .catch((error) => {
+        // Handle errors (e.g., display an error message)
+        console.error('Error deleting question:', error);
```

Comment:
```suggestion
        console.error('Error deleting question:', error);
        toast({
          title: 'Question deletion failed.',
          status: 'error',
          duration: 9000,
          isClosable: true,
        });
```

to display something to the user :P

`const toast = useToast()` somewhere